cluster/addons/dns/kube-dns/kube-dns.yaml.base

```diff
@@ -90,6 +90,17 @@ spec:
       securityContext:
         supplementalGroups: [ 65534 ]
         fsGroup: 65534
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: k8s-app
+                    operator: In
+                    values: ["kube-dns"]
+              topologyKey: kubernetes.io/hostname
```

Comment:
Did you run any scale test with that?
Note that OSS tests are using the default, which currently CoreDNS, so even if we merge it we won't know if it works or not.

This is a risky one - I think we were doing it at least 3 times in the past and always had to revert. I know that scheduling team put a ton of work into optimizing it, but I also don't know how good it is now.

I think the only way of answering this question would be to run our scale tests (actually density test would probably be enough, because we only care about scheduling throughput for this case - and it's much shorter - ~2.5h vs 10h of load test at 5k scale).
@mm4tt - FYI

Reply:
From a CoreDNS perspective, I did [open a PR to test the pod-anti-affinity](https://github.com/kubernetes/kubernetes/pull/84733) some time back, 
the scale tests seem to pass: https://github.com/kubernetes/kubernetes/pull/84733#issuecomment-553321221

Metrics from the test: https://github.com/kubernetes/kubernetes/pull/84733#issuecomment-553451336

@oxddr

Reply:
Hmm - the large tests were triggerred there, which is 2k-node cluster. I don't think that's enough - there can potentially be significant difference between 2k and 5k.

Reply:
We've run 5k-node scale tests against this change and it doesn't seem to cause any visible impact on pretty much any metrics.

So I'm fine with proceeding with this change from scalability perspective.
That said, please split this into two separate PRs (version bump and add of anti-affinity), as there is still a risk we may need to rollback it (we will be monitoring tests once that merges).

@prameshj ^^

Reply:
Ack, will do. Thanks for the update.

Reply:
I have split the version update in PR - https://github.com/kubernetes/kubernetes/pull/90744